crates/katana/core/src/metrics.rs

```diff
@@ -0,0 +1,14 @@
+//! Transaction pool metrics.
```

Comment:
We want this module to be the `Core metrics` more than the `Transaction pool metrics`.
In general from what `reth` is doing is having a metrics module per crate, which seems to be appropriate in our case too.